src/scaler/spreadscaler/mod.rs

```diff
@@ -394,10 +399,33 @@ fn compute_spread(spread_config: &SpreadScalerProperty) -> Vec<(Spread, usize)>
         // there was a case that I didn't imagine
         Ordering::Greater => {
             warn!("Requesting more actor instances than were specified");
-            spreads
+            computed_spreads
         }
-        Ordering::Equal => spreads,
+        Ordering::Equal => computed_spreads,
+    };
+
+    // case when spread_config.spread.is_empty() <=> requested_spreads.is_empty()
+    if computed_spreads.len() == 1 {
+        return computed_spreads;
     }
+
+    // spreads are returned in the order that they were declared in the config (ie, not by weight).
+    // this shouldn't matter; however, it is a safe behavior with respect to downstream processing.
+    let requested_spreads_indices: HashMap<String, usize> = spread_config
+        .spread
+        .iter()
+        .enumerate()
+        .map(|(idx, spread)| (spread.name.to_owned(), idx))
+        .collect();
+
+    computed_spreads.iter().fold(
+        vec![(Spread::default(), 0); computed_spreads.len()],
+        |mut output, spread| {
+            let idx = requested_spreads_indices.get(&spread.0.name).unwrap();
+            output[*idx] = spread.to_owned();
+            output
+        },
+    )
```

Comment:
I think I agree with your comment here that the spread shouldn't matter, and we probably don't need this extra processing step. This is only an internal method to the scaler and won't modify the order that we store the manifest in the kvstore, which is to say the user won't know we're doing this computation.

If you'd like to keep it, the only thing I would request is that the last `fold` is made safer without the use of an `unwrap` and direct Vec indexing 🙂

Reply:
fair point. my intuition is that the (re)ordering should be kept at least for the time being. Ergo, i will address the potentially unsafe stuff you mentioned. good call